src/controllers/media.ts

```diff
@@ -119,24 +156,48 @@ export const getBySanitizedTitle = async(ctx: Context): Promise<MediaMetadataInt
     throw new Error('title is required');
   }
 
-  let dbMeta: MediaMetadataInterface = await MediaMetadata.findOne({ title }).lean();
-
-  if (dbMeta) {
-    return ctx.body = dbMeta;
+  // If we already have a result, return it
+  const existingResultFromSearchMatch: MediaMetadataInterface = await MediaMetadata.findOne({ searchMatches: { $in: [title] } }).lean();
+  if (existingResultFromSearchMatch) {
+    return ctx.body = existingResultFromSearchMatch;
```

Comment:
I'm wondering if we should be filtering out the `searchMatches` from the Metadata document the API returns. They will get pretty large. Since it is a public API it might be strange being able to see a bunch of user filenames there. I guess they can't be attributed to any user however

Reply:
Perhaps that could be achieved with a Mongoose pre or post hook so we don't need to worry about filtering all the responses. Untested, but in a findOne hook we could add a projection to the query `{searchMatches: 0}`